db/migrate/20160301205721_change_elements_page_association_to_page_version.rb

```diff
@@ -0,0 +1,27 @@
+class ChangeElementsPageAssociationToPageVersion < ActiveRecord::Migration
+  def up
+    remove_index :alchemy_elements, [:page_id, :parent_element_id]
+    rename_column :alchemy_elements, :page_id, :page_version_id
+    add_index :alchemy_elements, [:page_version_id, :parent_element_id],
+      name: 'alchemy_elements_page_version_parent_element_idx'
+
+    Alchemy::Element.find_each do |element|
+      next unless element.page
+      version = element.page.public_version || element.page.current_version
+      element.update(page_version: version)
+      say "Assign element #{element.id} to #{version.id}"
```

Comment:
Please move this logic and the PageVersion logic into an Upgrader. We have to tell people that we're 
deleting and invalidating the cache and resetting all pages to a state as if they were recently published. It's a sorry behaviour, but exactly what we're fixing her. 

Also, there is a lot of model code being called from here, which we have a much easier time maintaining if it's only supposed to be used on a certain Alchemy version.